docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,209 @@
+        'post_type'    => 'post',
+      )
+    );
+
+    // Return the clean up function.
+    return function() use ( $post_id ) {
+      // Remove the test post created.
+      wp_delete_post( $post_id );
+    }
+  }
+}
+```
+
+### Preparation Classes
+
+Preparation classes are individual classes created to handle specific preparation logic in a single place.
+
+The [`Force_Single_Plugin_Preparation`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Preparations/Force_Single_Plugin_Preparation.php) in the Plugin Checker is an example of a preparation class.
+
+Preparation classes are useful for implementing the same preparation logic across multiple checks using the concept of shared preparations.
+
+### Shared Preparations
+
+Shared preparations are used to prevent running the same preparations multiple times.
+
+Before running checks against a plugin all shared preparations are collected and processed to remove any shared preparations that are the same. 
+
+Check classes can define the shared preparations the use by using the `With_Shared_Preparation` interface.
+
+The Check class should then implement the get_shared_preparations() method defined by the interface. This method returns an map of shared preparations where the preparation class name is the key and an array of constructor parameters as the value.
+
+Below is an example of how the `Enqueued_Scripts_Size_Check` uses shared preparations.
+
+```php
+  /**
+   * Returns an array of shared preparations for the check.
+   *
+   * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+   *               need any constructor arguments, it would just be an empty array.
+   */
+  public function get_shared_preparations() {
+    $demo_posts = array_map(
+      function( $post_type ) {
+        return array(
+          'post_title'   => "Demo {$post_type} post",
+          'post_content' => 'Test content',
+          'post_type'    => $post_type,
+          'post_status'  => 'publish',
+        );
+      },
+      $this->viewable_post_types
+    );
+
+    return array(
+      Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+    );
+  }
+```
+
+## Add the Check to the Plugin Checker
+
+In order to run the check as part of the Plugin Checker process it needs to be added to the Plugin Checkers list of available checks.
+
+This is done by using the `wp_plugin_check_checks` filter to register an instance of the check with its slug.
+
+- If you're contributing to the actual plugin checker, add the check to the list in the `Abstract_Check_Runner::register_checks()` method.
+- If you're implementing a check in code outside of the actual plugin checker, use the approach you've described here so far.
+
+```php
+add_filter(
+  'wp_plugin_check_checks',
+  function ( array $checks ) {
+    // Add the check to the map of all available checks.
+    $checks[ 'runtime_check' ] = new Runtime_Check();
+
+    return $checks;
+  }
+)
+```
```

Comment:
Please remove all this content and instead link to the other article. No need for all of this to be duplicated. You can simply say something like "Adding a runtime check works similarly to adding a static check.", and link to the other article.

Reply:
Thanks @felixarntz I've removed and linked to the relevant section